subcell_analysis/compression_analysis.py

```diff
@@ -159,3 +142,175 @@ def get_total_fiber_twist(
+            all_points.loc[fiber_at_time.index, metric] = get_asymmetry_of_peak(
+                fiber_values
+            )
+
+        if metric == "sum_bending_energy":
+            fiber_values = fiber_at_time[
+                ["xpos", "ypos", "zpos", "segment_energy"]
+            ].values
+            all_points.loc[fiber_at_time.index, metric] = get_sum_bending_energy(
+                fiber_values
+            )
+
+        if metric == "non_coplanarity":
+            fiber_values = fiber_at_time[["xpos", "ypos", "zpos"]].values
+            all_points.loc[fiber_at_time.index, metric] = get_third_component_variance(
+                fiber_values
+            )
+
+        if metric == "average_perp_distance":
+            fiber_values = fiber_at_time[["xpos", "ypos", "zpos"]].values
+            all_points.loc[
+                fiber_at_time.index, metric
+            ] = get_average_distance_from_end_to_end_axis(fiber_values)
+
+        if metric == "total_fiber_twist":
+            fiber_values = fiber_at_time[["xpos", "ypos", "zpos"]].values
+            all_points.loc[fiber_at_time.index, metric] = get_total_fiber_twist(
+                fiber_values
+            )
+
+        if metric == "energy_asymmetry":
+            fiber_values = fiber_at_time[
+                ["xpos", "ypos", "zpos", "segment_energy"]
+            ].values
+            all_points.loc[fiber_at_time.index, metric] = get_energy_asymmetry(
+                fiber_values
+            )
+    return all_points
+
+
+def run_compression_workflow(
+    all_points: pd.core.frame.DataFrame, metrics_to_calculate: list
+) -> pd.core.frame.DataFrame:
+    """
+    Calculates chosen metrics from cytosim output of fiber positions and properties across timesteps
+
+    Parameters
+    ----------
+    all_points: [(num_timepoints * num_points) x n columns] pandas dataframe
+        all_points is a dataframe of cytosim outputs that is generated after some post-processing.
+        includes [fiber_id, x_pos, y_pos, z_pos, xforce, yforce, zforce, segment_curvature, force_magnitude, segment_energy] columns and any metric columns
+    metrics: [n] list of metrics to calculate with metrics passed in as strings
+
+    Returns
+    -------
+    all_points dataframe with chosen metrics appended as columns
+
+    """
+    for metric in metrics_to_calculate:
+        all_points = run_metric_calculation(all_points, metric)
+    return all_points
+
+
+def get_sum_bending_energy(
+    fiber_energy: np.ndarray,
+) -> float:
+    """
+    Returns the sum bending energy given a single fiber x,y,z positions and segment energy values
+
+    Parameters
+    ----------
+    fiber_energy: [n x 4] numpy array
+        array containing the x,y,z positions of the polymer trace and segment energy
+        at a given time
+
+    Returns
+    -------
+    total_energy: float
+        energy of a vector at a given time
+    """
+    return np.sum(fiber_energy[3])
```

Comment:
We don't need to pass the fiber positions for this one as they aren't used in the energy calculations. This could just be a one liner in the `run_metric_calculation` function.

```suggestion

```